Tests/FunctionalTests/MiscellaneousTests.swift

```diff
@@ -491,6 +491,54 @@ class MiscellaneousTestCase: XCTestCase {
         }
     }
 
+    func testReportingErrorFromGitCommand() throws {
+        fixture(name: "Miscellaneous/MissingDependency") { prefix in
+
+            let fakeGit = prefix.appending(components: "bin", "fakegit")
+
+            try localFileSystem.createDirectory(fakeGit.parentDirectory)
+
+            // Write out fake git.
+            let stream = BufferedOutputByteStream()
+            stream <<< "#!/bin/sh" <<< "\n"
+            stream <<< "set -e" <<< "\n"
+            stream <<< "printf \"Some Error Reported Here\" " <<< "\n"
+            stream <<< "exit 555" <<< "\n"
+            try localFileSystem.writeFileContents(fakeGit, bytes: stream.bytes)
+
+            // Make it executable.
+            _ = try Process.popen(args: "chmod", "+x", fakeGit.asString)
```

Comment:
This is fine but we can make an utility method in TestSupport to create executable scripts. Something like:

```swift
createExectutableScript(at: fakeGit) { stream in
    stream <<< "..."
}
```